node/test/util/rm.js

```diff
@@ -0,0 +1,396 @@
+                expect: "x=S:I README.md",
+            },
+            "rm a clean submodule by name": {
+                initial: "x=S:C2-1 x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                paths: ["x/y/z"],
+                expect: "x=S:C2-1 x/y/z=foo;W x/q/z=bar;I x/y/z;Bmaster=2",
+            },
+            "rm a clean submodule by name --cached": {
+                initial: "x=S:C2-1 x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                paths: ["x/y/z"],
+                cached: true,
+                expect:
+                "x=S:C2-1 x/y/z=foo;W x/q/z=bar,x/y/z=foo;I x/y/z;Bmaster=2",
+            },            
+            "rm a clean file by removing its containing dir, no --recursive": {
+                initial: "x=S:C2-1 x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                paths: ["x/y"],
+                fails: true,
+                expect: "x=S:C2-1 x/y/z=foo;W x/q/z=bar;Bmaster=2",
+            },
+            "rm a clean file by removing its containing dir": {
+                initial: "x=S:C2-1 x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                paths: ["x/y"],
+                recursive: true,
+                expect: "x=S:C2-1 x/y/z=foo;W x/q/z=bar;I x/y/z;Bmaster=2"
+            },
+            "rm two clean files by removing their containing dir": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                paths: ["x/y"],
+                recursive: true,
+                expect:
+                "x=S:C2-1 x/y/a=foo,x/y/z=foo;W x/q/z=bar;" +
+                    "I x/y/a,x/y/z;Bmaster=2"
+            },
+            "rm two clean files by removing their grandparent dir": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                paths: ["x"],
+                recursive: true,
+                expect:
+                "x=S:C2-1 x/y/a=foo,x/y/z=foo;W x/q/z=bar;" +
+                    "I x/y/a,x/y/z;Bmaster=2"
+            },
+            "rm a non-existent thing": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                paths: ["z"],
+                fails: true,
+                expect: "x=S:C2-1 x/y/a=foo,x/y/z=foo;W x/q/z=bar;Bmaster=2",
+            },
+            "rm a submodule": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;C3-1 d=S/baz.git:2;" +
+                    "Bmaster=3;Bsub=2",
+                paths: ["d"],
+                expect: "x=S:C2-1 x/y/a=foo,x/y/z=foo;C3-1 d=S/baz.git:2;" +
+                    "I d;" +
+                    "Bmaster=3;Bsub=2",
+            },
+            "rm an open submodule": {
+                initial: "sub=S:C8-1 x/y/a=foo,x/y/z=foo;Bmaster=8|" +
+                    "x=S:C2-1 d=Ssub:8;" +
+                    "Bmaster=2;Od",
+                paths: ["d"],
+                expect: "sub=S:C8-1 x/y/a=foo,x/y/z=foo;Bmaster=8|" +
+                    "x=S:C2-1 d=Ssub:8;" +
+                    "Bmaster=2;I d"
+            },
```

Comment:
Similarly, if you leave a repo out from the expected case, it's assumed to be unchanged, so this `expect` case can be simplified to `x=I d`.